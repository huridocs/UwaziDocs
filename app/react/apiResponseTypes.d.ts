import { UserGroupSchema } from 'shared/types/userGroupType';
import { UserSchema } from 'shared/types/userType';
import {
  PreserveConfig,
  Settings,
  SettingsFilterSchema,
  SettingsLinkSchema,
} from 'shared/types/settingsType';
import { LanguageSchema } from 'shared/types/commonTypes';

export interface GroupMemberSchema {
  refId: string;
  username: string;
}

export interface ClientUserGroupSchema extends Omit<UserGroupSchema, '_id' | 'members'> {
  _id?: string;
  members: GroupMemberSchema[];
}

export interface ClientUserSchema extends Omit<UserSchema, '_id' | 'groups'> {
  _id?: string;
  groups?: {
    _id: string;
    name: string;
  }[];
}

export interface ClientSettingsFilterSchema extends Omit<SettingsFilterSchema, '_id'> {
  _id?: string;
}

export interface ClientSublink extends Omit<ClientSettingsLinkSchema, 'sublinks'> {}

export interface ClientSettingsLinkSchema extends Omit<SettingsLinkSchema, '_id' | 'sublinks'> {
  _id?: string;
  sublinks?: ClientSublink[];
}

export interface ClientPreserveConfig extends Omit<PreserveConfig, 'config'> {
  config: {
    token: string;
    template: string;
    user?: string;
  }[];
}

export interface ClientLanguageSchema extends Omit<LanguageSchema, '_id'> {
  _id?: string;
}

export interface ClientSettings
  extends Omit<Settings, '_id | filters | links | features | languages'> {
  _id?: string;
  filters?: ClientSettingsFilterSchema[];
  languages?: ClientLanguageSchema[];
  links?: ClientSettingsLinkSchema[];
  features?: Omit<Settings['features'], 'preserve | metadataExtraction'> & {
    preserve?: ClientPreserveConfig;
    metadataExtraction?: {
      url: string;
      templates?: {
        template: string;
        properties: string[];
      }[];
    };
    [k: string]: unknown | undefined;
  };
}
