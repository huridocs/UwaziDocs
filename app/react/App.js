import React from 'react';
import { Provider } from 'react-redux';
import { routes } from './Routes';
import CustomProvider from './App/Provider';
import { store } from './store';

const App = () => (
  <Provider store={store}>
    <CustomProvider>{routes}</CustomProvider>
  </Provider>
);

export default App;
