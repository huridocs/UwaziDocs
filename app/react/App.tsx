import React from 'react';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { getRoutes } from './Routes';
import CustomProvider from './App/Provider';
import { store } from './store';

const reduxState = store?.getState();

const router = createBrowserRouter(
  getRoutes(reduxState?.settings.collection.toJS(), reduxState?.user.get('_id'))
);

const App = () => (
  <Provider store={store as any}>
    <CustomProvider>
      <RouterProvider router={router} fallbackElement={null} />
    </CustomProvider>
  </Provider>
);

export { App };
