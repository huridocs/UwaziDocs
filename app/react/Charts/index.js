import loadable from '@loadable/component';

import colorScheme from './utils/colorScheme';
import arrayUtils from './utils/arrayUtils';
import ExtendedTooltip from './components/ExtendedTooltip';

const Bar = loadable(
  async () => import(/* webpackChunkName: "LazyLoadBar" */ './components/Bar.js')
);

const ColoredBar = loadable(
  async () => import(/* webpackChunkName: "LazyLoadColoredBar" */ './components/ColoredBar.js')
);

const Pie = loadable(
  async () => import(/* webpackChunkName: "LazyLoadPie" */ './components/Pie.js')
);

const StackedDualBarChart = loadable(
  async () =>
    import(
      /* webpackChunkName: "LazyLoadStackedDualBarChart" */ './components/StackedDualBarChart.js'
    )
);

export { Bar, ColoredBar, ExtendedTooltip, Pie, StackedDualBarChart, colorScheme, arrayUtils };
