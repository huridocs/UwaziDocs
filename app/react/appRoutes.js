import { getRoutes } from './Routes';
import { settingsAtom, atomStore, userAtom } from './V2/atoms';

export const routes = getRoutes && getRoutes(atomStore.get(settingsAtom), atomStore.get(userAtom)?._id);

