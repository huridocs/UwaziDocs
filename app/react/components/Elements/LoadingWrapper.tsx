import React from 'react';
import { Loader } from './Loader';

type LoadingWrapperProps = {
  isLoading: boolean;
  children?: React.ReactNode;
};

const LoadingWrapper = ({ isLoading, children }: LoadingWrapperProps) => {
  if (isLoading) {
    return <Loader />;
  }
  return <>{children}</>;
};

export { LoadingWrapper };
