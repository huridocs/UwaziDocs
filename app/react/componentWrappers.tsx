/* eslint-disable comma-spacing */
/* eslint-disable react/no-multi-comp */
import React, { useContext } from 'react';
import {
  useLocation,
  useMatches,
  useNavigate,
  useOutlet,
  useParams,
  useSearchParams,
} from 'react-router-dom';
import { AppMainContext } from './App/AppMainContext';

const withRouter =
  <T,>(Component: React.ComponentClass<T & { params?: any }, any>) =>
  (props: T & { params?: any }) => {
    const location = useLocation();
    const navigate = useNavigate();
    const params = useParams();
    const matches = useMatches();
    const [searchParams, setSearchParams] = useSearchParams();
    return (
      <Component
        {...props}
        navigate={navigate}
        location={location}
        params={{ ...params, ...(props.params ? props.params : {}) }}
        matches={matches}
        searchParams={searchParams}
        setSearchParams={setSearchParams}
      />
    );
  };

const withContext =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const mainContext = useContext(AppMainContext);
    return <Component {...props} mainContext={mainContext} />;
  };

const withOutlet =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const outlet = useOutlet(AppMainContext);
    return <Component {...props} outlet={outlet} />;
  };
export { withRouter, withContext, withOutlet };
