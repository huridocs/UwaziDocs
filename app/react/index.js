import React from 'react';
import ReactDOM from 'react-dom';

import * as Sentry from '@sentry/react';
import { BrowserTracing } from '@sentry/tracing';

import { AppContainer } from 'react-hot-loader';

import { SENTRY_APP_DSN } from 'app/config';
import App from './App';

import './App/sockets';

if (SENTRY_APP_DSN) {
  Sentry.init({
    dsn: SENTRY_APP_DSN,
    integrations: [new BrowserTracing()],
    tracesSampleRate: 0.1,
  });
}

const render = Component => {
  ReactDOM.hydrate(
    <AppContainer>
      <Component />
    </AppContainer>,
    document.getElementById('root')
  );
};

render(App);

if (module.hot) {
  module.hot.accept('./App', () => {
    const nextApp = require('./App');
    render(nextApp);
  });
}
