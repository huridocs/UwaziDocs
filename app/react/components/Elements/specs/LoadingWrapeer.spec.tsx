/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { LoadingWrapper } from '../LoadingWrapper';

describe('LoadingWrapper', () => {
  it('should display a loading bar when the loading prop is set to true', () => {
    const { container } = render(<LoadingWrapper isLoading />);
    expect(container).toMatchSnapshot();
  });

  it.each([
    'Simple text',
    <p>Simple HTML</p>,
    [<h1 key="1">Multiple children</h1>, <p key="2">second child</p>],
    undefined,
  ])('should render children when loading prop is set to false', children => {
    const { container } = render(<LoadingWrapper isLoading={false}>{children}</LoadingWrapper>);
    expect(container).toMatchSnapshot();
  });
});
