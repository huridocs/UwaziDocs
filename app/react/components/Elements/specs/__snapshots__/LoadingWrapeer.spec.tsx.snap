// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWrapper should display a loading bar when the loading prop is set to true 1`] = `
<div>
  <div
    class="cs-loader"
  >
    <div
      class="cs-loader-inner"
    >
      <label>
         ●
      </label>
      <label>
         ●
      </label>
      <label>
         ●
      </label>
    </div>
  </div>
</div>
`;

exports[`LoadingWrapper should render children when loading prop is set to false 1`] = `
<div>
  Simple text
</div>
`;

exports[`LoadingWrapper should render children when loading prop is set to false 2`] = `
<div>
  <p>
    Simple HTML
  </p>
</div>
`;

exports[`LoadingWrapper should render children when loading prop is set to false 3`] = `
<div>
  <h1>
    Multiple children
  </h1>
  <p>
    second child
  </p>
</div>
`;

exports[`LoadingWrapper should render children when loading prop is set to false 4`] = `<div />`;
